src/TextInput.js

```diff
@@ -1,28 +1,34 @@
-import React from 'react'
 import classnames from 'classnames'
 import PropTypes from 'prop-types'
 import systemPropTypes from '@styled-system/prop-types'
-import styled from 'styled-components'
-import {layout} from 'styled-system'
+import styled, {css} from 'styled-components'
+import {variant, layout} from 'styled-system'
 import {COMMON, get} from './constants'
 import theme from './theme'
 
-function TextInputBase({autocomplete, theme, size, block, className, ...rest}) {
-  const classes = classnames(className, 'form-control', {
-    'input-block': block,
-    'input-sm': size === 'small',
-    'input-lg': size === 'large'
-  })
-  const inputProps = {
-    className: classes,
-    autoComplete: autocomplete,
-    type: 'text',
-    ...rest
+const sizeVariants = variant({
+  variants: {
+    small: {
+      minHeight: '28px',
+      px: 2,
+      py: '3px',
+      fontSize: 0,
+      lineHeight: '20px'
+    },
+    large: {
+      px: 2,
+      py: '10px',
+      fontSize: 3
+    }
   }
-  return <input {...inputProps} />
-}
+})
 
-const TextInput = styled(TextInputBase)`
+const TextInput = styled.input.attrs(({autocomplete, theme, variant, block, className, ...rest}) => ({
```

Comment:
Let's take this opportunity to remove the `autocomplete` prop and ask users to use the standard `autoComplete` React prop - them we don't need to "rename" them down on line 28 and we're following React standards

Reply:
Perfect, I was wondering about that :)